docs/guides/lambda.rst

```diff
@@ -0,0 +1,65 @@
+*****************************************
+The Trouble (Or Lack Thereof) With Lambda
+*****************************************
+
+Lambda expressions are a common and useful part of the Python programming language.
+However, there is one problem with them: syntactically, they do not allow for type
+annotations. While it is perfectly simple to write ``lambda x: x``, you cannot directly
+indicate a type for x. (Type annotations are indicated by a colon, and so is the end
+of the lambda parameter list. Where would the type annotation go?)
+
+However, despite this infelicity, lambda expressions are not immune from static typing,
+and in fact follow the same static type rules as everything else. Type checkers try
+to deduce the type of the lambda arguments and return value, and if they can't they
+fall back to ``Any``. Due to the inability to directly indicate types for these,
+``Any`` tends to pop up quite often here. This means that many type errors may occur
+here unnoticed, which is bad. For instance, the following example is a runtime type
+error, but is uncaught in most (perhaps all) type checkers:
+
+.. code-block:: python
+
+   f1 = lambda a, b: a * b
+   f1(1, "a")
+
+(The alternative way of writing this, ``(lambda a, b: a + b)(1, "a")``, is typically
+caught by type checkers, because it is simple and immediate enough that they are able
+to deduce that a type error will occur.)
+
+There are some workarounds to this problem, which all involve assigning the lambda to
+something, in one way or another, and annotating that. This is a bit unfortunate,
+because the idiomatic use of a lambda involves not doing that. In fact, at that point
+you might as well just define a normal function. Let's call that our first workaround.
+
+``def f(x: object) -> object: return x``
+
+The second workaround is equivalent: assigning the lambda to a variable, and annotating
+the type of that variable with a Callable.
+
+``f: Callable[[object], object] = lambda x: x``
+
+Type comments on function definitions do not actually work on lambda, nor do
+normal type comments help (although you can use a type comment on an assignment
```

Comment:
Might be better to not mention type comments? They're effectively deprecated, and in the context we're discussing I don't think they add anything over variable annotations.

Reply:
Incorporated; thanks!